From kay.sievers@vrfy.org  Thu Sep 14 02:27:10 2006
Subject: CONFIG_SYSFS_DEPRECATED
From: Kay Sievers <kay.sievers@vrfy.org>
To: greg@kroah.com
Date: Thu, 14 Sep 2006 11:23:28 +0200
Message-Id: <1158225809.3167.8.camel@min.off.vrfy.org>

Provide a way to support older versions of udev that are shipped in
older distros.  If this option is enabled, it will also turn off the
compatible symlinks in sysfs that older programs might rely on.


Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 init/Kconfig |   19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

--- gregkh-2.6.orig/init/Kconfig
+++ gregkh-2.6/init/Kconfig
@@ -249,6 +249,25 @@ config CPUSETS
 
 	  Say N if unsure.
 
+config SYSFS_DEPRECATED
+	bool "Create deprecated sysfs files"
+	help
+	  This option creates deprecated symlinks such as the
+	  "device"-link, the <subsystem>:<name>-link, and the
+	  "bus"-link. It may also add deprecated key in the
+	  uevent environment.
+	  None of these features or values should be used today, as
+	  they export driver core implementation details to userspace
+	  or export properties which can't be kept stable across kernel
+	  releases.
+
+	  If enabled, this option will also move any device structures
+	  that belong to a class, back into the /sys/class heirachy, in
+	  order to support older versions of udev.
+
+	  If you are using a distro that was released in 2006 or later,
+	  it should be safe to say N here.
+
 config RELAY
 	bool "Kernel->user space relay support (formerly relayfs)"
 	help
